tools/lib/h5tools_type.c

```diff
@@ -55,8 +55,12 @@ h5tools_get_little_endian_type(hid_t tid)
             break;
 
         case H5T_FLOAT:
-            if (size == 2)
-                p_type = H5Tcopy(H5T_IEEE_F16LE);
+            if (size == 2) {
+                if (true == H5Tequal(tid, H5T_IEEE_F16LE) || true == H5Tequal(tid, H5T_IEEE_F16BE))
```

Comment:
Had to make sure the correct type between float16 and bfloat16 is picked so that the data comes out correctly.

Reply:
Similarly, seems this could cause problems with custom floats in general. NO need to fix for this PR though.

Reply:
Yes, from the other comment there are several places, especially in the tools and Java code, where we should handle such types better. Handling of those types has been poor, even outside of my changes here which admittedly do introduce problems with specific custom 16-bit floating point formats.